src/components/Homepage/HomePage.jsx

```diff
@@ -1,61 +1,132 @@
-import React, { useEffect } from "react";
-import { useState } from "react";
-import { Link } from "react-router-dom";
-const axios = require("axios");
+import React, { Suspense, useEffect, useReducer } from "react";
+import { Link, useNavigate, useParams } from "react-router-dom";
+import axios from "axios";
+import Spinner from "../Spinner/Spinner";
+import ReactPaginate from "react-paginate";
 
 const api_key = process.env.REACT_APP_API_KEY;
 
-export default function HomePage({ setMovieInfo }) {
-  const [movies, setMovies] = useState([]);
+const movieDataActionTypes = {
+  SET_MOVIE_PAGE_DATA: "SET_MOVIE_PAGE_DATA",
+  SET_IS_LOADING: "SET_IS_LOADING",
+};
 
-  useEffect(() => {
-    fetchData();
-  }, []);
+export default function HomePage() {
+  const [state, dispatch] = useReducer(reducer, {
+    movies: [],
+    totalCountPages: 500,
+    currentPage: 1,
+    isLoading: false,
+  });
 
-  async function fetchMovieData(id) {
-    const response = await axios.get(
-      `https://api.themoviedb.org/3/movie/${id}`,
-      {
-        params: {
-          api_key: api_key,
-        },
+  const navigate = useNavigate();
+  const params = useParams();
+
+  function reducer(state, action) {
```

Comment:
Please move reducer to a separate file in the same folder.

Reply:
Component should be as clean as possible. If we can - we shall move function ouside.

Reply:
fixed